### PR TITLE
authproxy.md: strip X-Remote-User

### PR DESCRIPTION
**Pull request by stapelberg**
follow-up for https://github.com/coreos/dex/pull/1100

### CODE REVIEW

**Comment by ericchiang** [comment]
@stapelberg can you make a change to dex's code as well? I'd like for dex's server to remove any X-Remote-Header values sent to `/callback` and not `/callback/{connector}`

**Comment by stapelberg** [comment]
Sure. Done.

File: server/server.go [hunk 1]
```diff
@@ -240,7 +240,12 @@ func newServer(ctx context.Context, c Config, rotationStrategy rotationStrategy)
 	handleWithCORS("/keys", s.handlePublicKeys)
 	handleFunc("/auth", s.handleAuthorization)
 	handleFunc("/auth/{connector}", s.handleConnectorLogin)
-	handleFunc("/callback", s.handleConnectorCallback)
+	r.HandleFunc(path.Join(issuerURL.Path, "/callback"), func(w http.ResponseWriter, r *http.Request) {
+		// Strip the X-Remote-User header to prevent security issues on
+		// misconfigured authproxy connector setups.
+		r.Header.Del("X-Remote-User")
```
Comment (ericchiang) [inline server/server.go]: Let's future proof this against other headers:

```go
for key := range r.Header {
    if strings.HasPrefix(strings.ToLower(key), "x-remote-") {
        r.Header.Del(key)
    }
}
```

Comment (stapelberg) [reply server/server.go]: Done.

**Pull request merged**